core/json/src/test/scala/net/liftweb/json/JsonPrintingSpec.scala

```diff
@@ -34,6 +36,77 @@ object JsonPrintingSpec extends Specification  with JValueGen with ScalaCheck {
     forAll(rendering)
   }
 
+  "rendering special double values by default" should {
+    "render a standard double as is" in {
+      val double = Random.nextDouble
+      JsonAST.compactRender(JDouble(double)) must_== double.toString
+    }
+
+    "render positive infinity as null" in {
+      JsonAST.compactRender(JDouble(Double.PositiveInfinity)) must_== "null"
+    }
+
+    "render negative infinity as null" in {
+      JsonAST.compactRender(JDouble(Double.NegativeInfinity)) must_== "null"
+    }
+
+    "render NaN as null" in {
+      JsonAST.compactRender(JDouble(Double.NaN)) must_== "null"
+    }
+  }
+
+  "rendering special double values with as-is handling" should {
+    def render(json: JValue) = {
+      JsonAST.render(
+        json,
+        JsonAST.RenderSettings(0, doubleRenderer = JsonAST.RenderSpecialValuesAsIs)
+      )
+    }
+
+    "render a standard double as is" in {
+      val double = Random.nextDouble
+      render(JDouble(double)) must_== double.toString
+    }
+
+    "render positive infinity as null" in {
+      render(JDouble(Double.PositiveInfinity)) must_== "Infinity"
+    }
+
+    "render negative infinity as null" in {
+      render(JDouble(Double.NegativeInfinity)) must_== "-Infinity"
+    }
+
+    "render NaN as null" in {
+      render(JDouble(Double.NaN)) must_== "NaN"
+    }
+  }
+
+  "rendering special double values with special value exceptions enabled" should {
+    def render(json: JValue) = {
+      JsonAST.render(
+        json,
+        JsonAST.RenderSettings(0, doubleRenderer = JsonAST.FailToRenderSpecialValues)
+      )
+    }
+
+    "render a standard double as is" in {
+      val double = Random.nextDouble
+      render(JDouble(double)) must_== double.toString
+    }
+
+    "render positive infinity as null" in {
+      render(JDouble(Double.PositiveInfinity)) must throwAn[IllegalArgumentException]
+    }
+
+    "render negative infinity as null" in {
```

Comment:
Should read something like `throw an exception if attempting to render negative infinity`